frontend/src/hooks/useTimezoneFormatter/useTimezoneFormatter.ts

```diff
@@ -0,0 +1,98 @@
+import { Timezone } from 'components/CustomTimePicker/timezoneUtils';
+import dayjs from 'dayjs';
+import timezone from 'dayjs/plugin/timezone';
+import utc from 'dayjs/plugin/utc';
+import { useCallback, useEffect, useMemo } from 'react';
+
+// Initialize dayjs plugins
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
+// Types
+export type TimestampInput = string | number | Date;
+interface CacheEntry {
+	value: string;
+	timestamp: number;
+}
+
+//
+
+// Constants
+const CACHE_SIZE_LIMIT = 1000;
+const CACHE_CLEANUP_PERCENTAGE = 0.5; // Remove 50% when limit is reached
+
+function useTimezoneFormatter({
+	userTimezone,
+}: {
+	userTimezone: Timezone;
+}): {
+	formatTimezoneAdjustedTimestamp: (
+		input: TimestampInput,
+		format?: string,
+	) => string;
+} {
+	// Initialize cache using useMemo to persist between renders
+	const cache = useMemo(() => new Map<string, CacheEntry>(), []);
+
+	// Clear cache when timezone changes
+	useEffect(() => {
+		cache.clear();
+	}, [cache, userTimezone]);
```

Comment:
**Thought:** The cache object is stable (as per useMemo), so it doesn't need to be in the dependency array. This could result in unnecessary re-clearing of the cache.

Reply:
IMO, it won't cause any re-clearing of the cache, since it's memoized. if we remove the cache from dependencies array, the ESLint is going to complain.